applications/nrf_desktop/doc/dvfs.rst

```diff
@@ -0,0 +1,104 @@
+DVFS is critical for balancing performance and power consumption, especially in battery-operated devices.
+
+In the nRF Desktop, the DVFS module enables the system to use the lowest possible frequency and voltage when the application performs less computation-challenging activities such as Bluetooth LE advertising.
+However, if performing more computation-consuming tasks nRF Desktop DVFS module will dynamically change to a higher frequency to increase performance.
+
+Requesting for frequency change is done by requesting clock from clock control module.
+Clock control module is responsible for choosing the right clock frequency and voltage.
+The DVFS is set by the sysctrl core and Clock control module requests are send through nrfs service DVFS.
+
+.. note::
+   In the current solution, it is assumed that no other module requests frequency changes.
+   The nRF Desktop DVFS module is the sole entity responsible for managing frequency and voltage adjustments.
+
+Supported devices
+*****************
+
+The module is only available for nrf54h20 SoC.
+Each device has a different set of available operating points.
+
+For example the nrf54h20 SoC has the following available operating points (in documentation called frequency):
+
+1. DVFS_FREQ_HIGH - high frequency - 320 MHz, 0.8 V
+#. DVFS_FREQ_MEDLOW - medium-low frequency - 128 MHz, 0.6V
+#. DVFS_FREQ_LOW - low frequency - 64 MHz, 0.5V
+
+Module events
+*************
+
+.. include:: event_propagation.rst
+    :start-after: table_dvfs_start
+    :end-before: table_dvfs_end
+
+.. note::
+    |nrf_desktop_module_event_note|
+
+Configuration
+*************
+
+The nRF Desktop Dynamic Voltage and Frequency Scaling (DVFS) module is enabled by default on supported board targets and can be disabled by setting the :option:`CONFIG_DESKTOP_DVFS` Kconfig option to ``n``.
+
+The module can be configured using the following Kconfig options:
+
+The nRF Desktop DVFS module defines independent states that have different requirements for the DVFS operating point.
+The module listens for Even Manager events that are sent and turns state on/off according to the event.
+It support the following DVFS states:
+
+* ``DVFS_STATE_INITIALIZING`` - State indicating that core initialization is in process.
+* ``DVFS_STATE_LLPM_CONNECTED`` - State indicating that Bluetooth Low Latency Packet Mode is active.
+* ``DVFS_STATE_USB_CONNECTED`` - State indicating that USB is connected.
+* ``DVFS_STATE_CONFIG_CHANNEL`` - State indicating that Config Channel is active.
+* ``DVFS_STATE_SMP_TRANSFER`` - State indicating that SMP transfer is active.
+
+Each DVFS state can be configured using the following Kconfig options:
+
+* ``CONFIG_DESKTOP_DVFS_STATE_<STATE>_ENABLE`` - Enable DVFS state <STATE>.
+* ``CONFIG_DESKTOP_DVFS_STATE_<STATE>_ACTIVE_FREQ`` - Choice of active frequency of state <STATE>.
```

Comment:
```suggestion
* ``CONFIG_DESKTOP_DVFS_STATE_<STATE>_ENABLE`` - Enables DVFS state <STATE>.
* ``CONFIG_DESKTOP_DVFS_STATE_<STATE>_ACTIVE_FREQ`` - Sets the active frequency for state <STATE>.
```

Reply:
Just wondering, but is <STATE> how this should be indicated? Should we express it graphically in some way, e.g., ``<STATE>``? What do you think @peknis ?

Reply:
Can we mention that the choice defines frequency `requested` by the state? Or minimal frequency required when the state is active?